File: resources/capability-bundle/.claude/skills/deep-research/scripts/synthesize-evidence.py
import json
import sys


payload = json.loads(sys.argv[1] if len(sys.argv) > 1 else "{}")
claims = payload.get("claims") if isinstance(payload.get("claims"), list) else []
allowed_states = {"supported", "disputed", "inferred", "unknown"}

normalized = []
for index, claim in enumerate(claims):
    claim = claim if isinstance(claim, dict) else {}
    status = claim.get("status")
    normalized.append({
        "claim_id": str(claim.get("claim_id", f"claim-{index + 1}")),
        "text": str(claim.get("text", "")).strip(),
        "source_refs": [str(value) for value in claim.get("source_refs", [])]
        if isinstance(claim.get("source_refs"), list) else [],
        "status": status if status in allowed_states else "unknown",
        "as_of": None if claim.get("as_of") is None else str(claim["as_of"]),
    })

sys.stdout.write(json.dumps({"claims": normalized}, ensure_ascii=False, separators=(",", ":")))

File: resources/capability-bundle/.claude/skills/deep-research/scripts/synthesize-evidence.mjs
const payload = JSON.parse(process.argv[2] ?? '{}');
const claims = Array.isArray(payload.claims) ? payload.claims : [];

const normalized = claims.map((claim, index) => ({
  claim_id: String(claim.claim_id ?? `claim-${index + 1}`),
  text: String(claim.text ?? '').trim(),
  source_refs: Array.isArray(claim.source_refs) ? claim.source_refs.map(String) : [],
  status: ['supported', 'disputed', 'inferred', 'unknown'].includes(claim.status)
    ? claim.status
    : 'unknown',
  as_of: claim.as_of == null ? null : String(claim.as_of),
}));

process.stdout.write(JSON.stringify({ claims: normalized }));
